### ISSUE TITLE
Homie ESP-8266 is not working without wifi

### ISSUE THREAD

**Issue by kmehran** [homieiot/homie-esp8266#536]
[...template with no change deleted...]

**Comment by kmehran** [comment]
I designed a wifi light-switch, based on Homie and Openhab Server, the first important function of 
each light switch is turning on and off the lights, but it cannot turn on/off the light during wifi/mqtt
connection or even when one or both of them lost.
One of the Most important matter about adding extra functionality or new technology to an old 
system (like Light Switch) is do not affect its basic function(s).

**Comment by Alex9779** [comment]
I found this somewhere else here, maybe in another issue description, the solution is depending on your implementation that you run the loop where you handle the events the manual switch raises in the mail loop in addition to the Homie.loop().
Here an excerpt of one of my implementations:
```
void loop()
{  
  Homie.loop();

  button.tick();
}
```
In this firmware is use the lib "OneButton" for handling a button, it is a uplight control firmware and the uplight has one button where you can turn it on and off in addtion to control it with Homie.

**Comment by kmehran** [comment]
Dear Alexander,

That should be my solution, thanks for your
complete answer.

Regards,
Kaveh.

**Comment by euphi** [comment]
See here, how to change the Homie-ESP8266 sources to get your desired behavior:

https://github.com/homieiot/homie-esp8266/pull/490/files

https://github.com/homieiot/homie-esp8266/blob/4e1cfe583c1e067928f874ee7ef7e04b3dd4c8d0/src/Homie/Boot/BootNormal.cpp

**Comment by Misiu** [comment]
@euphi, as this is labeled with `Feature_V3`, does it means that this will be added to develop-v3 branch?
I'm trying to build Wemos D1 based multi-relay switch and recently I found this repo. Because OpenHab support homie 3.x I want to use this library.
As many wrote before sometimes we don't want to wait for MQTT to connect at the start, the most important thing is the working relay.

I've found an example that I want to use as a base for my sketch - https://github.com/homieiot/homie-esp8266/blob/develop-v3/examples/IteadSonoffButton/IteadSonoffButton.ino

How should I change it so that I can toggle replay when I don't have MQTT connected?
Example showing such behavior would be awesome.
